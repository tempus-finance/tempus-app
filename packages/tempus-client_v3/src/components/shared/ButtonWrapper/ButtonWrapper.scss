.tc__btn {
  padding: 0;
  border: none;
  background-color: transparent;
  cursor: pointer;

  &:disabled {
    cursor: default;
  }
}
