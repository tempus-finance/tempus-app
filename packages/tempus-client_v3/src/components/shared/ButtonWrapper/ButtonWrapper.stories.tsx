import { ComponentStory } from '@storybook/react';
import React from 'react';

import ButtonWrapper from './ButtonWrapper';

export default {
  title: 'ButtonWrapper',
  component: ButtonWrapper,
  argTypes: {},
};

const style = {
  background: 'rgba(0, 0, 0, 0.1)',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  padding: '10px',
};

const Template: ComponentStory<typeof ButtonWrapper> = args => (
  <div style={style}>
    <ButtonWrapper {...args}>My Button Wrapper</ButtonWrapper>
  </div>
);

export const Primary = Template.bind({});
Primary.args = {
  title: 'primary',
};
