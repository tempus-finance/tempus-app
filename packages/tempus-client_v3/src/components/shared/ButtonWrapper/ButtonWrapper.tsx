import React, { FC, forwardRef, memo, HTMLProps } from 'react';

import './ButtonWrapper.scss';

export interface ButtonWrapperProps {
  className?: string;
  title?: string;
  disabled?: boolean;
  selected?: boolean;
  onClick?: () => void;
}

const Button = forwardRef<HTMLButtonElement, ButtonWrapperProps>((props, ref) => {
  const { children, className = '', title, disabled = false, selected = false, onClick } = props;

  return (
    <button
      ref={ref}
      className={`tc__btn ${className}`}
      title={title}
      disabled={disabled}
      onClick={onClick}
      data-selected={selected}
    >
      {children}
    </button>
  );
});

export default memo(Button as FC<ButtonWrapperProps & HTMLProps<HTMLButtonElement>>);
