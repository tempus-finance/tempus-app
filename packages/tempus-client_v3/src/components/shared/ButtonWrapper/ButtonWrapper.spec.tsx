import { fireEvent, render } from '@testing-library/react';
import ButtonWrapper, { ButtonWrapperProps } from './ButtonWrapper';

const mockOnClick = jest.fn();

const defaultProps: ButtonWrapperProps = {
  title: 'my title',
  onClick: mockOnClick,
};

const subject = (props: ButtonWrapperProps) => render(<ButtonWrapper {...props}>My Button</ButtonWrapper>);

describe('Button', () => {
  it('renders a button with a title attribute', () => {
    const { queryByTitle, queryByText } = subject(defaultProps);

    const actualText = queryByText('My Button');
    const actualLabel = queryByTitle('my title');

    expect(actualText).not.toBeNull();
    expect(actualLabel).not.toBeNull();

    expect(actualText).toMatchSnapshot();
  });

  it('calls `onClick` when the button is clicked', () => {
    const { getByRole } = subject(defaultProps);

    const button = getByRole('button');
    fireEvent.click(button);

    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });
});
