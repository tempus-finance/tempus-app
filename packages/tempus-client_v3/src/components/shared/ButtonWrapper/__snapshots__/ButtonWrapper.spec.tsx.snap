// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders a button with a title attribute 1`] = `
<button
  class="tc__btn "
  data-selected="false"
  title="my title"
  type="button"
>
  My Button
</button>
`;
